[16.0][FIX] product_multi_company: don't change rules, but respect previous company assignments

yajo opened the pull request:
Remove lots of code thanks to https://github.com/OCA/multi-company/pull/760.

Also, call the new `fill_company_ids` hook, to respect company assignments done before the module installation.

Draft until:

- [x] https://github.com/OCA/multi-company/pull/760

**Instructions for functional tests:** This refactor is internal and should not be noticed functionally. The module `product_multi_company` should work just as always. If so, then this is good.

@moduon MT-8873

yajo committed:
[FIX] product_multi_company: don't change rules, but respect previous company assignments

Remove lots of code thanks to https://github.com/OCA/multi-company/pull/760.

Also, call the new `fill_company_ids` hook, to respect company assignments done before the module installation.

@moduon MT-8873

pedrobaeza reviewed (commented):
Shouldn't you include here migration scripts the same as in `partner_multi_company`?

yajo commented:
You're right, thanks for the review. Script added.

pedrobaeza reviewed (approved):
/ocabot merge major

Please fw-port it.

pedrobaeza commented:
@yajo please check the CI

PieterPaulussen commented:
@yajo The tests for this module are broken. Please check https://github.com/OCA/multi-company/actions/runs/13632724910/job/38103974812#step:8:847. Are you available to fix this? 
Additionally, Pedro asks to forward port these fixes to v17.0 too.

LoisRForgeFlow commented:
@yajo Just a kind reminder here. Migration to 18.0 is already in progress, and it is probably better if we can include this change.

yajo committed:
[FIX] base_multi_company: don't eliminate company_id value

Some forms can send both values, but in that case the one from company_id shouldn't be just dropped. It should be kept instead.

@moduon MT-8873

yajo commented:
Thanks, fixed. Sorry I have a long delay in attending notifications.

yajo commented:
Please could you merge? This one is approved and tests are already fixed. Thanks!

pedrobaeza commented:
/ocabot merge major

LoisRForgeFlow commented:
@yajo I have notified in the v18 migration, I guess it is still needed the fwport to v17, right?

yajo commented:
We're skipping v17. I'm sorry about that 😅 

But anybody can do the fwport if needed.
